Book-Store-MVC/Controllers/BookController.cs

```diff
@@ -14,84 +14,47 @@ namespace Book_Store_MVC.Controllers
 {
     public class BookController : Controller
     {
-        private readonly IGenericRepository<Book> genericRepository;
+        private readonly BookRepository  genericRepository;
         private readonly IGenericRepository<Category> catgenericRepository;
         private readonly IGenericRepository<Author> aurhrepo;
-
-        private readonly BookStoreContext bookStore;
         private readonly IMapper mapper;
-       
-
-        //public BookController(BookStoreContext bookStore , IMapper mapper )
-        //{
-        //    this.bookStore = bookStore;
-        //    this.mapper = mapper;
-        //}
+        private readonly IGenericRepository<Models.Publisher> publisherrepo;
 
-
-        public BookController(IGenericRepository<Book> genericRepository, IGenericRepository<Category> catgenericRepository, IGenericRepository<Author> aurhrepo ,  IMapper mapper ,
-          BookStoreContext bookStore)
+        public BookController(BookRepository genericRepository, IGenericRepository<Category> catgenericRepository, IGenericRepository<Author> aurhrepo ,  IMapper mapper ,
+         IGenericRepository<Models.Publisher> publisherrepo )
         {
             this.genericRepository = genericRepository;
             this.catgenericRepository = catgenericRepository;
             this.aurhrepo = aurhrepo;
             this.mapper = mapper;
-         
-            this.bookStore = bookStore;
+            this.publisherrepo = publisherrepo;
+        
         }
 
 
         #region Index
-        public ActionResult Index(int id, string Search)
+        public ActionResult Index(int id = 0, string searchTerm = null, int pageNumber = 1, int pageSize = 10)
         {
-            // var bookowithnoinclude = genericRepository.GetAll().AsQueryable();
-            var books = genericRepository.GetAll(b => b.Category, b => b.Author, b => b.Publisher).ToList();
-
+            List<Book> books = genericRepository.GetAll(id, searchTerm, pageNumber , pageSize).ToList(); 
```

Comment:
getAll return one page only

Reply:
in the view "index " of book  the list item increase the pages with one page  depending on the count of data and pages 
try to add more books and see if it works

Reply:
regarding pagination
i need the total to be counted in the GetAll method in order to use it in the controller